framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -863,55 +908,62 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
             {
                 if (replay_result == XR_SUCCESS)
                 {
-                    // Add it to the list of received events
-                    received_events_.push_back(*out_eventData);
+                    // If this was a valid event, but not the one we're interested in, record
+                    // it to a list of received events for us to use later.
+                    previously_received_unhandled_events_.push_back(*out_eventData);
                     GFXRECON_LOG_WARNING("Recording event for later %s (0x%x, %u)",
                                          GetEventTypeString(capture_event->type),
                                          capture_event->type,
                                          capture_event->type);
 
                     // If we grow too lare on the event vector, it's probably because we have
-                    // received a bunch of events we can not handle.  So remove the first 100
+                    // received a bunch of events we can not handle.  So remove the first 300
                     // events to make room for more without bloating the list size.
                     // TODO: Perhaps do this more elegantly?
-                    if (received_events_.size() > 1000)
+                    if (previously_received_unhandled_events_.size() > 1000)
```

Comment:
1000 and 300 should be const statics of the ReplayConsumer

Reply:
and maybe like 5000, and 500?

Reply:
I switched to constants.  And I set them to 2000 and 500.  I don't like it to get too big because they are large structures.